lib/execution-info.js

```diff
@@ -0,0 +1,526 @@
+'use strict';
+
+const utils = require('./utils');
+const types = require('./types');
+const errors = require('./errors');
+
+/**
+ * A base class that represents a wrapper around the user provided query options with getter methods and proper
+ * default values.
+ */
+class ExecutionInfo {
```

Comment:
Maybe i am nitpicking too much but `ExecutionInfo`  does not feel descriptive enough about what it is.  These are really options / configuration and "Info" to me implies that it's metadata/informational.  What do you think about `ExecutionOptions` or `ExecutionConfig`?    Another possibility is referring to things as a `Request` i.e. `RequestOptions` / `RequestConfig`.

`ExecutionInfo` could also be seen as overloaded as other drivers ([c#](https://docs.datastax.com/en/latest-csharp-driver-api/html/T_Cassandra_ExecutionInfo.htm), [java](https://docs.datastax.com/en/drivers/java/3.6/com/datastax/driver/core/ExecutionInfo.html)).

Reply:
I think its a valid concern, specially given that `ExecutionInfo` is already taken by other drivers.

`ExecutionOptions` is a nice alternative: no collision and clear meaning. I'll change it.